### ISSUE TITLE
As an admin I want to edit a specific customer’s information so that I can ensure that the details are up to date.

### ISSUE THREAD

**Issue by IsabelHaataja** [KYH-Net23/IdentityUserService#33]
Authors: Kimmo Ahola & Isabel Haataja

## Acceptance Criteria:
- [ ] Admins should receive the customer information from a database.
- [ ] Admins can save the changes to a database.
- [ ] The user account type (admin, customer) can not be updated.
- [ ] Changes should not be saved if the fields are invalid.
- [ ] Error messages should be displayed if the user tries to save incorrect information.
- [ ] There can not be duplicates of Email.

## Tasks:
- [ ] KYH-Net23/Rika#171
- [ ] Create a form with fields for editing the customers information, include validation.
- [ ] Create a "Save" button.
- [ ] Create a “Cancel” button.
- [x] #22
- [ ] #23
- Name
- Email
- Birthdate
- Gender
- Date Changed

## Activity Diagram: 
![Image](https://github.com/user-attachments/assets/fd2e3747-c0c4-4cd6-bf89-3d58770533b4)

## Test Cases:

- [ ] Case 1:
Prerequisites: 
The user is logged in as an admin
The admin has navigated to the customer whose information is supposed to be edited
**Steps:** 
1. Press the “Edit” button
2. Choose an information field to edit
3. Write new information
4. Press “Save”
**Expected Results:** The field that has been edited is replaced by the new information, a success message appears.

- [ ] Case 2:
Prerequisites: Admin tries to duplicate an email.
**Steps:**
1. Update the email field to a value that already exists in the database.
2. Attempt to save changes.
**Expected Results:** The system displays a message that the email is already taken.

- [ ] Case 3:
Prerequisites:
**Steps:**
1. Leave a/several fields blank
2. Attempt to save the changes.
**Expected Results:** The system prevents changes and displays an error message.

- [ ] Case 4:
Prerequisites: Admin tries to cancel while editing.
**Steps:**
1. Update a field
2. Press “Cancel”
**Expected Results:** The system does not save any changes made.

## Sequence Diagram:
![Image](https://github.com/user-attachments/assets/cda4cac5-f029-411c-984c-c4a581017c81)
